src/main/kotlin/nonblocking/ReactorSelectorManager.kt

```diff
@@ -0,0 +1,82 @@
+package nonblocking
+
+import kotlinx.coroutines.*
+import java.io.Closeable
+import java.nio.channels.CancelledKeyException
+import java.nio.channels.SelectableChannel
+import java.nio.channels.SelectionKey
+import java.nio.channels.Selector
+import java.util.concurrent.ConcurrentHashMap
+import kotlin.coroutines.Continuation
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+class ReactorSelectorManager : CoroutineScope, Closeable {
+    override val coroutineContext: CoroutineContext = Dispatchers.Default + CoroutineName("ReactorSelectorManager")
+
+    private val selector: Selector = Selector.open()
+    private val selectionKeysToContinuation: ConcurrentHashMap<SelectableChannel, ConcurrentHashMap<Int, Continuation<Unit>>> =
```

Comment:
Could you tell me when we're removing channels from the map?

Reply:
I will change this map to Attachment object